### PR TITLE
[Refactor](predicate) Unify Conditions and ColumnPredicate

### LINKED ISSUE #12009: [Enhancement] Unify Conditions and ColumnPredicate
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Unify Conditions and ColumnPredicate

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #12009

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/block_column_predicate.h [hunk 1]
```diff
@@ -57,6 +61,16 @@ class BlockColumnPredicate {
 
     virtual void evaluate_vec(vectorized::MutableColumns& block, uint16_t size,
                               bool* flags) const {};
+
+    virtual bool evaluate_and(const std::pair<WrapperField*, WrapperField*>& statistic) const {
+        DCHECK(false) << "should not reach here";
```
Comment (BiteTheDDDDt) [inline be/src/olap/block_column_predicate.h:66]: Maybe we can just use `LOG(FATAL)` here?

File: be/src/olap/in_list_predicate.h [hunk 2]
```diff
@@ -181,6 +188,60 @@ class InListPredicateBase : public ColumnPredicate {
...
+            return true;
+        }
+    }
+
+    bool evaluate_and(const segment_v2::BloomFilter* bf) const override {
+        if constexpr (PT == PredicateType::IN_LIST) {
+            for (auto value : _values) {
+                bool existed = false;
+                if constexpr (std::is_same_v<T, StringValue>) {
+                    existed = bf->test_bytes(value.ptr, value.len);
```
Comment (BiteTheDDDDt) [inline be/src/olap/in_list_predicate.h:226]: maybe we can remove `existed` and  return directly

File: be/src/olap/comparison_predicate.h [hunk 3]
```diff
@@ -199,6 +201,101 @@ class ComparisonPredicateBase : public ColumnPredicate {
         _evaluate_bit<true>(column, sel, size, flags);
     }
 
+    bool evaluate_and(const std::pair<WrapperField*, WrapperField*>& statistic) const override {
+        if (statistic.first == nullptr || statistic.second == nullptr) {
```
Comment (BiteTheDDDDt) [inline be/src/olap/comparison_predicate.h:205]: Does we can move those check out of this function? looks like a lot of duplicate code

File: be/src/olap/comparison_predicate.h [hunk 4]
```diff
@@ -199,6 +201,101 @@ class ComparisonPredicateBase : public ColumnPredicate {
...
+                T tmp_min_uint32_value = 0;
+                memcpy((char*)(&tmp_min_uint32_value), statistic.first->cell_ptr(),
+                       sizeof(uint24_t));
+                T tmp_max_uint32_value = 0;
+                memcpy((char*)(&tmp_max_uint32_value), statistic.second->cell_ptr(),
+                       sizeof(uint24_t));
+                return tmp_min_uint32_value <= _value && tmp_max_uint32_value >= _value;
+            } else if constexpr (std::is_same_v<T, StringValue>) {
+                auto min = reinterpret_cast<const Slice*>(statistic.first->cell_ptr());
+                auto max = reinterpret_cast<const Slice*>(statistic.second->cell_ptr());
```
Comment (BiteTheDDDDt) [inline be/src/olap/comparison_predicate.h:222]:
```cpp
struct Slice {
public:
    char* data;
    size_t size;
}
struct StringValue {
    char* ptr;
    size_t len;
}
```
Does we can use `reinterpret_cast` directly?

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**